bundle.go

```diff
@@ -120,39 +123,60 @@ func (b SortBundlesByVersion) Len() int           { return len(b) }
 func (b SortBundlesByVersion) Swap(i, j int)      { b[i], b[j] = b[j], b[i] }
 func (b SortBundlesByVersion) Less(i, j int) bool { return b[i].Version < b[j].Version }
 
+type SortBundlesByTime []Bundle
+
+func (b SortBundlesByTime) Len() int           { return len(b) }
+func (b SortBundlesByTime) Swap(i, j int)      { b[i], b[j] = b[j], b[i] }
+func (b SortBundlesByTime) Less(i, j int) bool { return b[i].Time.UnixNano() < b[j].Time.UnixNano() }
+
 // ValidateBundles is a plain validation type for a list of version bundles. A
 // list of version bundles is exposed by authorities. Lists of version bundles
 // of multiple authorities are aggregated and grouped to reflect distributions.
 type ValidateBundles []Bundle
 
+func (b ValidateBundles) Copy() ValidateBundles {
+	raw, err := json.Marshal(b)
+	if err != nil {
+		panic(err)
+	}
+
+	var copy ValidateBundles
+	err = json.Unmarshal(raw, &copy)
+	if err != nil {
+		panic(err)
+	}
+
+	return copy
+}
+
 func (b ValidateBundles) Validate() error {
+	if len(b) == 0 {
+		return microerror.Maskf(invalidBundlesError, "version bundles must not be empty")
+	}
+
 	if b.hasDuplicatedVersions() {
-		return microerror.Mask(invalidBundleError)
+		return microerror.Maskf(invalidBundlesError, "version bundle versions must be unique")
+	}
+
+	b1 := b.Copy()
+	b2 := b.Copy()
+	sort.Sort(SortBundlesByTime(b1))
+	sort.Sort(SortBundlesByVersion(b2))
+	if !reflect.DeepEqual(b1, b2) {
+		return microerror.Maskf(invalidBundlesError, "version bundle versions must always increment")
```

Comment:
I don't get what this means. We make two copies, sort one by time, one by version, and if they're not the same? Why would they be the same?

Reply:
Consider an operator exposing its version bundles. 

```
[
  {
    ...
    "time": "2017-08-17T00:00:00Z",
    "version": "0.1.0",
    ...
  },
  {
    ...
    "time": "2017-08-18T00:00:00Z",
    "version": "0.0.9",
    ...
  },
]
```

There are two bundles. The first one was there first (in time). Its version is `0.1.0`. As time goes by we should only add new version bundles where we increment versions. The second version bundle of the example above came in later (in time), but its version is lower as the first one. This must never happen. The code you commented on tried to ensures this.

Reply:
I get it. Nice.